src/components/ContentItem.jsx

```diff
@@ -34,17 +31,16 @@ function ContentItem({recipe}) {
             setUserFavorites(favorites)
             setLiked(true)
         }
-        
     }
 
-    const checkLiked =()=>{
+    const checkLiked = useCallback(()=>{
         const ids = userFavorites ? userFavorites.map(recipe=>{return recipe.id}) : []
         ids.includes(recipe.id) ? setLiked(true) : setLiked(false)
-    }
+    },[recipe,userFavorites])
 
     useEffect(()=>{
         checkLiked()
-    },[])
+    },[checkLiked])
```

Comment:
I believe the intent of this `useEffect()` is to instantiate `checkLiked()` whenever this component is rendered. To do that you don't need to attach an _Array Dependency_. 

The code as is, is saying: there's a dependency on `checkLiked()`, if checkLiked() changes then run the useEffect. Well, checkLiked isn't a state of anything, so there's no identifier to it that's there's been a change. You want _Array Dependencies_ to be related to some form of state. The useEffect uses the dependency and only runs the logic within itself, when the dependency has made a change.

IE. Have a DOM element with an onClick eventListener `toggleChecked`: `<div onClick='toggleChecked'>Click It</div>' Now we can write some logic with a useEffect like this:
```
const [isChecked, setIsChecked] = useState(false);

const toggleChecked = () => {
  setChecked(!isChecked);
}

useEffect(()=> {
    // this will alert user every time toggleChecked is clicked
   alert('user has set checked to' + checked);
}, [isChecked]);
```

Reply:
updated that to your suggestion.